lib/stdlib/doc/src/uri_string_usage.xml

```diff
@@ -0,0 +1,323 @@
+    </p>
+    <pre>
+    3> uri_string:parse("http://%6C%6Fcal%23host/%F6re%26bro%20").
+    #{host => "%6C%6Fcal%23host",path => "/%F6re%26bro%20",
+      scheme => "http"}}
+    </pre>
+    <p>The input is a valid URI but how can you decode those
+    percent-encoded octets? You can try to normalize the input with
+    <c>uri_string:normalize/1</c>. The normalize operation decodes those
+    percent-encoded triplets that correspond to a character in the unreserved set.
+    Normalization is a safe, idempotent operation that converts a URI into its
+    canonical form:</p>
+    <pre>
+    4> uri_string:normalize("http://%6C%6Fcal%23host/%F6re%26bro%20").
+    "http://local%23host/%F6re%26bro%20"
+    5> uri_string:normalize("http://%6C%6Fcal%23host/%F6re%26bro%20", [return_map]).
+    #{host => "local%23host",path => "/%F6re%26bro%20",
+      scheme => "http"}
+    </pre>
+    <p>There are still a few percent-encoded triplets left in the output. At this point,
+    when the URI is already parsed, it is safe to apply application specific decoding on
+    the remaining character triplets. OTP provides a function,
+    <c>uri_string:percent_decode/1</c> for raw percent decoding
+    that you can use on the host and path components, or on the whole map:
+    </p>
+    <pre>
+    6> uri_string:percent_decode("local%23host").
+    "local#host"
+    7> uri_string:percent_decode("/%F6re%26bro%20").
+    <![CDATA[{error,invalid_utf8,<<"/öre&bro ">>}]]>
+    8> uri_string:percent_decode(#{host => "local%23host",path => "/%F6re%26bro%20",
+    scheme => "http"}).
+    <![CDATA[#{host => "local#host",
+      path => {error,invalid_utf8,<<"/öre&bro ">>},
+      scheme => "http"}]]>
+    </pre>
+    <p>The <c>host</c> was successfully decoded but the path contains at least one
+    character with
+    non-UTF-8 encoding. In order to be able to decode this, we have to make assumptions
+    about the encoding used in these triplets. The most obvious choice is
+    <i>latin-1</i>, so we can try
+    to transcode the path to UTF-8 and run the percent-decode operation on
+    the transcoded string:
+    </p>
+    <pre>
+    9> uri_string:transcode("/%F6re%26bro%20", [{in_encoding, latin1}]).
+    "/%C3%B6re%26bro%20"
+    10> uri_string:percent_decode("/%C3%B6re%26bro%20").
+    <![CDATA["/öre&bro "]]>
+    </pre>
+    <p>It is important to emphasize that it is not safe to apply
+    <c>uri_string:percent_decode/1</c> directly on an input URI:
+    </p>
+    <pre>
+    11> uri_string:percent_decode("http://%6C%6Fcal%23host/%C3%B6re%26bro%20").
+    <![CDATA["http://local#host/öre&bro "
+    12> uri_string:parse("http://local#host/öre&bro ").]]>
+    {error,invalid_uri,":"}
+    </pre>
+    <p>This is the reason why OTP depricated the <c>http_uri:decode/1</c> function.
+    It can change a URI or render it invalid. <c>http_uri:encode/1</c> is depricated
+    as well, encoding happens in <c>uri_string:recompose/1</c>.
```

Comment:
You should expand on this to better explain why there's no uri_string:encode equivalent.